packages/core/strapi/lib/services/entity-service/types/params/populate.d.ts

```diff
@@ -0,0 +1,45 @@
+import type { Attribute, Common, Utils } from '@strapi/strapi';
+
+/**
+ * Wildcard notation for populate
+ *
+ * To populate all the root level relations
+ */
+type WildcardNotation = '*';
+
+type PopulatableKeys<TSchemaUID extends Common.UID.Schema> = Utils.Guard.Never<
+  Attribute.GetPopulatableKeys<TSchemaUID>,
+  string
+>;
+
+/**
+ * Union of all possible string representation for populate
+ *
+ * @example
+ * type A = 'image'; // ✅
+ * type B = 'image,component'; // ✅
+ * type c = '*'; // ✅
+ * type D = 'populatableField'; // ✅
+ * type E = '<random_string>'; // ❌
+ */
+type StringNotation<TSchemaUID extends Common.UID.Schema> =
+  | WildcardNotation
+  | PopulatableKeys<TSchemaUID>
+  | `${string},${string}`
+  | `${PopulatableKeys}.${string}`;
+
+/**
+ * Array notation for populate
+ *
+ * @example
+ * type A = ['image']; // ✅
+ * type B = ['image', 'component']; // ✅
+ * type C = ['populatableField']; // ✅
+ * type D = ['<random_string>']; // ❌
+ */
+type ArrayNotation<TSchemaUID extends Common.UID.Schema> = StringNotation<TSchemaUID>[];
```

Comment:
`['someField', '*']` will be treated as valid, which might be ok, but it's also not in the documentation. Should wildcard only be available when used by itself and not in an array?

Reply:
Indeed, looking at the implementation, we're not really trying to couple both the wildcard & regular string. We could indeed exclude the wildcard from the array notation

Reply:
I mean it's possible and doesn't really throw an error

Reply:
we can't compeletely discard it from the array notation this ['*'] is still equivalent to this '*'

Reply:
I say go with however it's currently implemented. If `['someField', '*']` works to populate everything, this is fine.